src/main/java/gyro/azure/compute/VirtualMachineResource.java

```diff
@@ -978,7 +1255,18 @@ public void update(GyroUI ui, State state, Resource current, Set<String> changed
     public void delete(GyroUI ui, State state) {
         Azure client = createClient();
 
+        VirtualMachine virtualMachine = client.virtualMachines().getById(getId());
         client.virtualMachines().deleteById(getId());
+
+        if (getDeleteOsDiskOnTerminate()
+                && virtualMachine != null
+                && !"specialized".equals(getVmImageType())) {
+
+            ui.write("\n Deleting Dynamically created OS Disk (%s)", virtualMachine.osDiskId());
```

Comment:
Seems unnecessary to write output here. The diff should be clear what is going to be removed.

Also, indentation is different than other places that output is written.

Reply:
Gyro doesn't have any knowledge about this disk, so no output would be written when it is deleted. It is dynamically created with the VM depending on the Image Type. Should we create a resource to represent this? This would have to be added to the state after the VM is created.